### PR TITLE
Natspec comments

### PR DESCRIPTION
**Pull request by ritik99**

### CODE REVIEW

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 1]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
 
     /**
      * @dev Used to withdraw asset from Saving Account
-     * @param withdrawTo address to which asset should be sent
-     * @param amount amount of liquidity shares to withdraw
-     * @param asset address of the asset to be withdrawn
-     * @param strategy strategy from where asset has to withdrawn(ex:- compound,Aave etc)
-     * @param withdrawShares boolean indicating to withdraw in liquidity share or underlying token
+     * @param _withdrawTo address to which asset should be sent
+     * @param _amount amount of liquidity shares to withdraw
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:192]: This is the base token amount, we don't use liquidity shares anymore as input to savings account

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 2]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
     /**
      * @dev Used to switch saving strategy of an asset
-     * @param currentStrategy initial strategy of asset
-     * @param newStrategy new strategy to invest
-     * @param asset address of the asset
-     * @param amount amount of **liquidity shares** to be reinvested
+     * @param _currentStrategy initial strategy of asset
+     * @param _newStrategy new strategy to invest
+     * @param _asset address of the asset
+     * @param _amount amount of **liquidity shares** to be reinvested
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:151]: We use only base token amount as input

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 3]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
 
-        emit Withdrawn(msg.sender, withdrawTo, amountReceived, token, strategy);
+        emit Withdrawn(msg.sender, _withdrawTo, _amountReceived, _token, _strategy);
     }
 
+    /*
+    * @notice used to withdraw assets from a savings account strategy
+    * @param _from sender address
+    * @param _to receiver address
+    * @param _amount amount to be withdrawn
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:223]: Let's probably specify that this is base token amount

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 4]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
+        (_token, _amountReceived) = _withdraw(_to, _amount, _asset, _strategy, _withdrawShares);
 
-        emit Withdrawn(from, msg.sender, amountReceived, token, strategy);
+        emit Withdrawn(_from, msg.sender, _amountReceived, _token, _strategy);
     }
 
+    /*
+    * @notice invoked to perform the actual withdrawals
+    * @param _withdrawTo receiver address
+    * @param _amount amount to be withdrawn
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:259]: Let's probably specify that this is base token amount

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 5]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
+        allowance[msg.sender][_token][_to] = _amount;
 
-        emit Approved(token, msg.sender, to, amount);
+        emit Approved(_token, msg.sender, _to, _amount);
     }
 
     function approveFromToCreditLine(
-        address token,
+        address _token,
         address from,
```
Comment (prateekreddy) [inline contracts/SavingsAccount/SavingsAccount.sol:353]: Let's add _ to other params as well

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/SavingsAccount/SavingsAccount.sol [hunk 6]
```diff
@@ -54,319 +54,368 @@ contract SavingsAccount is ISavingsAccount, Initializable, OwnableUpgradeable {
...
 
         strategyRegistry = _strategyRegistry;
     }
 
+    /*
+    * @notice called to deposit assets into a savings account strategy
+    * @param amount amount to be transferred
+    * @param asset asset to be transferred
+    * @param strategy strategy into which assets should be deposited
+    * @param to address of the savings account owner
```
[the remark this answers is not in the document]
Comment (ritik99) [reply contracts/SavingsAccount/SavingsAccount.sol:80]: savings account holder?

**Comment by prateekreddy** [comment]
Naspec docs completed as a separate task. Closing as this is not needed anymore.

**Pull request closed without merging**